src/PrestaShopBundle/Form/Admin/Type/CategoryCoverImageType.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This form type is used as a container of sub forms, each sub form will be rendered as a part of an accordion.
+ */
+class CategoryCoverImageType extends FileType
+{
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        parent::buildView($view, $form, $options);
+        $view->vars['download_url'] = $options['download_url'];
```

Comment:
I don't know if you were aware of it but we have introduced a `download_url` on all `FileType` already via the `DownloadFileExtension` so it should already work and be handled via the form UI kit theme. Since you have a custom form theme for those elements you won't be able to use the generic form theme, but at least the option should already be handled and passed down to the view so you can simplify these types a bit

Reply:
It complains about not having download_url if I don't include it, even though i am extending FileType. 

I guess i could try to redefine downloadable_file to extend my new custom type but I don't know if that makes sense.

```
  form.extension.image_with_preview_downloadable_file:
    class: 'PrestaShopBundle\Form\Admin\Extension\DownloadFileExtension'
    tags:
      - { name: form.type_extension, extended_type: Symfony\Component\Form\Extension\Core\Type\FileType }
```